android/java/org/chromium/chrome/browser/crypto_wallet/presenters/Amount.java

```diff
@@ -0,0 +1,81 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.crypto_wallet.presenters;
+
+import android.text.TextUtils;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+public class Amount {
+    private BigDecimal mValue;
+
+    public Amount(BigDecimal value) {
+        if (value == null) {
+            this.mValue = BigDecimal.ZERO;
+            return;
+        }
+        this.mValue = value;
+    }
+
+    public Amount(String value) {
+        if (TextUtils.isEmpty(value)) {
+            this.mValue = BigDecimal.ZERO;
+            return;
+        }
+        this.mValue = new BigDecimal(value);
+    }
+
+    public Amount(double value) {
+        this.mValue = new BigDecimal(value);
+    }
+
+    public Amount add(BigDecimal value) {
+        BigDecimal safeVal = safeVal(value);
+        mValue = mValue.add(safeVal);
+        return this;
+    }
+
+    public Amount add(Amount value) {
+        BigDecimal safeVal = safeVal(value);
+        mValue = mValue.add(safeVal);
+        return this;
+    }
+
+    // Default max decimal precision is 8
+    public String toStringFormat() {
+        return toStringFormat(8);
+    }
+
+    public String toStringFormat(int decimalLength) {
+        return mValue.setScale(decimalLength, RoundingMode.HALF_UP)
+                .stripTrailingZeros()
+                .toPlainString();
+    }
+
+    public String toHex() {
+        if (mValue.signum() == 0) return "0x0";
+        return "0x" + mValue.toBigInteger().toString(16);
+    }
+
+    private BigDecimal safeVal(BigDecimal value) {
+        safeInit();
```

Comment:
I don't get why do we need to call `safeInit` here if `mValue` and `value` are unrelated. That function has to static perhaps as well

Reply:
It is just an extra check to make sure the existing value is never null to avoid any potential exception. I made `mValue` private for time being but may change it to `public`.
We shouldn't/can't use static here because `mValue` is not static.

Reply:
we don't use `mValue` in `private BigDecimal safeVal(BigDecimal value) {` at all. That function doesn't access or doesn't modify anything from `Amount` class. Why it cannot be static?

Reply:
Practically, we can make any function static which does not use member function, but the amount class is not a helper Utility class so better to avoid static functions separate in a Utility class.

Reply:
So tell me please, why Amount class has to be created to do that?
```
    private BigDecimal safeVal(BigDecimal value) {
        return (value == null) ? BigDecimal.ZERO : value;
    }
```
why do we need this pointer accessible to some unrelated class at all to that function.
I will be ok to be it like that if you convince me that Amount class creation is necessary for that function execution.

Reply:
If someone requires to use a particular method of a class then it can be converted to static though pragmatically, it can cause issues like:
- We won't be able to use other methods of amount class if we are directly using the safeVal
- It's a private method which can be changed anytime so shouldn't be directly accessible
I hope it make sense.

Reply:
no it's not, I can say for any helper function in Utils that I need it to be inside some particular class as it can be changed anytime. I don't see anything why that class is `private` and cannot be used outside of that class and why it's not static and keeps a reference on `this` pointer on some unrelated class.

Reply:
and could you please give an answer to that question `why Amount class has to be created to do that?`

Reply:
1. Pragmatically, we shouldn't make a method public if we are not sure or it can be modified.
2. As I have said earlier, can make it static if needed but better to keep it in a Utils class of static method to keep similar things together,

`why Amount class has to be created to do that?`
Usage of any class is not forced but the purpose of a class is to offer similar features.

Let's ask the opinions from others: @samartnik @AlexeyBarabash @deeppandya

Reply:
use `private static` for such functions and it won't be accessible directly from other places

Reply:
updated https://github.com/brave/brave-core/blob/54e2cbc2217f1a43e1340bafd9a39b869dac6e6b/android/java/org/chromium/chrome/browser/crypto_wallet/presenters/Amount.java#L64

Reply:
@Pavneet-Sing 
I would also use `static` for  `safeVal` as it doesn't use any members. Basically, when I see the static method in Java or C++, I instantly understand that the method doesn't use access to members. 

About to put or not into Utils class - in my opinion, static methods often are by itself something like class-internals-util method. So private static seems good.

If `safeVal(BigDecimal value)` will be required to be used outside of `Amount` then move it to `Utils`.